frankenstein/chunk_store.go

```diff
@@ -383,18 +388,24 @@ func (c *AWSChunkStore) lookupChunks(userID string, from, through model.Time, ma
 	return chunkSets, nil
 }
 
+func next(s model.LabelName) model.LabelName {
+	l := len(s)
+	return s[:l-2] + model.LabelName(s[l-1]+1)
+}
+
 func (c *AWSChunkStore) lookupChunksFor(userID string, hour int64, metricName model.LabelValue, matchers []*metric.LabelMatcher, incomingChunkSets chan []wire.Chunk, incomingErrors chan error) {
 	var chunkSets [][]wire.Chunk
 	for _, matcher := range matchers {
-		// TODO build support for other matchers
-		if matcher.Type != metric.Equal {
-			incomingErrors <- fmt.Errorf("%s matcher not supported yet", matcher.Type)
-			return
-		}
 
 		hashValue := hashValue(userID, hour, metricName)
-		rangeMinValue := rangeValue(matcher.Name, matcher.Value, minChunkID)
-		rangeMaxValue := rangeValue(matcher.Name, matcher.Value, maxChunkID)
+		var rangeMinValue, rangeMaxValue []byte
+		if matcher.Type == metric.Equal {
+			rangeMinValue = rangeValue(matcher.Name, matcher.Value, minChunkID)
+			rangeMaxValue = rangeValue(matcher.Name, matcher.Value, maxChunkID)
+		} else {
+			rangeMinValue = rangeValue(matcher.Name, "", minChunkID)
+			rangeMaxValue = rangeValue(next(matcher.Name), "", maxChunkID)
```

Comment:
I realize you're already manually filtering out chunks below that don't have the right label name because the inclusive BETWEEN behavior may include wrong ones. 

However, wouldn't it be better to use `minChunkID` here, to get as few as possible of the wrong chunks selected in the range?